homeassistant/components/light/mochad.py

```diff
@@ -67,8 +70,15 @@ def brightness(self):
 
     def _get_device_status(self):
         """Get the status of the light from mochad."""
-        with mochad.REQ_LOCK:
-            status = self.light.get_status().rstrip()
+        from pymochad import device
+
+        if self._controller.ctrl and self._controller.connect_event.wait():
+            light = device.Device(self._controller.ctrl, self._address,
+                                  comm_type=self._comm_type)
+            with mochad.REQ_LOCK:
```

Comment:
We shouldn't use sync locks in our thread pool. If we really need a lock it should be async. In that case use a coroutine that can use an async lock and schedule the current sync I/O on the executor thread pool with `hass.async_add_executor_job`. Save the lock in `hass.data` and pass it to each entity.

```py
async def _async_get_device_status(self):
    async with self._update_lock:
        await self.hass.async_add_executor_job(self._get_device_status)

async def async_update(self):
    await self._async_get_device_status()
```